Added Licensing documentation (#179)
Mirantis/mke-docs: content/docs/getting-started/obtain-license.md

---
title: Licensing MKE4
weight: 2
---

{{< callout type="warning" >}}

You must have a valid license to lawfully run MKE4. For more
information, refer to [Mirantis Agreements and Terms](https://legal.mirantis.com/).

{{< /callout >}}

> After you [Install the MKE image](install-MKE-CLI.md), proceed with downloading
> your MKE license as described below. This section also contains steps to apply
> your new license using the MKE web UI and configuration file.
> 
> ## Obtain your MKE4 license
> 

Reviewer: ```suggestion
## Obtain your MKE4 license

[Install the MKE image](install-MKE-CLI.md) prior to downloading your MKE4 license.
```

```

   A new tab opens in your browser.

5. Click **View file** to download your license file.

{{< callout type="info" >}}

Though MKE4 is generally a subscription-only service, you can obtain a free trial license from Mirantis. Make your request using the [Mirantis contact form](https://www.mirantis.com/contact).

{{< /callout >}}

## Set the license

1. Log in to the MKE web UI with an administrator account.
2. In the left-side navigation panel, navigate to **Admin Settings** -> **License**.
3. Insert the license into the **license key** field. To do this, click
   **Choose File** and navigate to the ``.lic`` file you previously downloaded.
   Alternately, you can copy-paste the contents of the ``.lic`` file into the
   field as a text string.

   ![Add a license](img/add-a-license.png)

4. Click **Save settings** to update the MKE4 license.

## Set the license in the configuration

1. Set the license under ``spec.license.token`` in the MKE4
   configuration file:

    ```yaml
    spec:
      license:
        token: <your-license-file>
    ```

2. Apply the license:

   ```commandline
   mkectl apply
   ```

> 3. Check the license status:
>    resource:

Reviewer: ```suggestion
3. Check the license status:
```

---

> title: Obtain the license

Reviewer: ```suggestion
title: Licensing MKE4
```

weight: 2
---

After you [Install the MKE image](install-MKE-CLI.md), proceed with downloading
your MKE license as described below. This section also contains steps to apply
your new license using the MKE web UI and configuration file.

{{< callout type="warning" >}}

Users are not authorized to run MKE without a valid license. For more
information, refer to [Mirantis Agreements and Terms](https://legal.mirantis.com/).

{{< /callout >}}

## Download your MKE license

> 1. Open an email from Mirantis Support with the subject Welcome to Mirantis’
>    CloudCare Portal and follow the instructions for logging in.
> 
> 2. If you did not receive the CloudCare Portal email, it is likely that you
>    have not yet been added as a Designated Contact. To remedy this, contact
>    your Designated Administrator.

Reviewer: ```suggestion
1. Locate the _Welcome to Mirantis' CloudCare Portal _email sent to you from Mirantis
Support. If you do not have the email, confirm with your Designated Administrator
that you have added as a Designated Contact.
```

> 3. In the top navigation bar, click **Environments**.

Reviewer: ```suggestion
2. Click **Environments** in the top navigation bar of the MKE web UI.
```

> 4. Click the **Cloud Name** associated with the license you want to download.

Reviewer: ```suggestion
3. Click the **Cloud Name** that is associated with the license you want to download.
```

> 5. Scroll down to **License Information** and click the **License File URL**. 

Reviewer: ```suggestion
4. Scroll down to **License Information** and click the **License File URL**. 
```

>    A new tab opens in your browser.

Reviewer: Is it a new tab that opens? Or a dialog window? Also, what is the name of the tab/window? Assuming it has a name of some sort, the sentence should read "The <name-of> tab/dialog will open."

> 6. Click **View file** to download your license file.

Reviewer: ```suggestion
5. Click **View file** to download your license file.
```

{{< callout type="info" >}}

> Though MKE is generally a subscription-only service, Mirantis offers a free
> trial license by request. Use our [contact form](https://www.mirantis.com/contact)
> to request a free trial license.

Reviewer: ```suggestion
Though MKE4 is generally a subscription-only service, you can obtain a free trial license from Mirantis. Make your request using the [Mirantis contact form](https://www.mirantis.com/contact).
```

{{< /callout >}}

> ## Set the license in MKE web UI

Reviewer: ```suggestion
## Set the license
```

Reviewer: To keep it simple, we should refer to the environment for each licensing step at the start of the sections and not in the headings.

> 1. Log in to your MKE instance using an administrator account.

Reviewer: ```suggestion
1. Log in to the MKE web UI with an administrator account.
```

> 2. In the left panel, navigate to **Admin Settings** -> **License**. 

Reviewer: ```suggestion
2. In the left-side navigation panel, navigate to **Admin Settings** -> **License**. 
```

> 3. Paste the token as a string into a **license key** text box, or click

Reviewer: Token? Modal?

I suggest editing this step to read:

"Insert the license into the **license key** field. To do this,  click **Choose File** and navigate to the .lic file you previously downloaded. Alternately, you can copy-paste the contents of the .lic file into the field as a text string."

   **Choose File** and select a license key (.lic) file from a modal.

   ![Add a license](img/add-a-license.png)

> 4. Click **Save settings**. MKE will automatically update with the new settings.

Reviewer: ```suggestion
4. Click **Save settings** to update the MKE4 license.
```

Reviewer: Nothing automatic about it, as the user has to click something to make it happen.

## Set the license in the configuration

> 1. Set the license token under ``spec.license.token`` in the MKE 4

Reviewer: ```suggestion
1. Set the license under ``spec.license.token`` in the MKE4
```

   configuration file:

    ```yaml
    spec:
      license:

>         token: <your-license-token>

Reviewer: ```suggestion
        token: <your-license-file>
```

    ```

2. Apply the license:

   ```commandline
   mkectl apply
   ```

> 3. Check the license status, users can see the status of the ``mkeconfig``

Reviewer: ```suggestion
3. Check the license status:
```
